src/convertToObject.js

```diff
@@ -6,7 +6,19 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const splittedString = sourceString
+    .split(';')
+    .map((str) => str.split(':'))
+    .map((str) => str.map((value) => value.trim()))
+    .map((str) => str.filter((value) => value !== ''))
+    .filter((str) => str.length !== 0)
+    .reduce((object, item) => {
+      object[item[0]] = item[1];
+
+      return object;
+    }, {});
+
+  return splittedString;
```

Comment:
The return statement correctly returns the result of the `reduce` operation. However, remember to update the variable name here as well if you change the `splittedString` variable name to something more descriptive.